nodestream_plugin_neptune/neptune_connector.py

```diff
@@ -0,0 +1,78 @@
+from nodestream.databases.copy import TypeRetriever
+from nodestream.databases.database_connector import (DatabaseConnector,
+                                                     QueryExecutor)
+
+from .ingest_query_builder import NeptuneIngestQueryBuilder
+from .neptune_connection import NeptuneAnalyticsConnection, NeptuneDBConnection
+from .neptune_query_executor import NeptuneQueryExecutor
+
+
+class NeptuneConnector(DatabaseConnector, alias="neptune"):
+    @classmethod
+    def from_file_data(
+        cls, mode: str, region: str, host: str = None, graph_id: str = None, **kwargs
+    ):
+        return cls(
+            mode=mode,
+            host=host,
+            graph_id=graph_id,
+            region=region,
+            async_partitions=kwargs.get("async_partitions"),
+            ingest_query_builder=NeptuneIngestQueryBuilder(),
+        )
+
+    def __init__(
+        self,
+        mode: str,
+        region: str,
+        async_partitions: int,
+        ingest_query_builder: NeptuneIngestQueryBuilder,
+        host: str = None,
+        graph_id: str = None,
+    ) -> None:
+        if mode != "database" and mode != "analytics":
+            raise ValueError("`mode` must be either 'database' or 'analytics'")
```

Comment:
minor: Generally, this is considered more pythonic. But I don't really care _too_ much if you prefer the other way. 

```suggestion
        if mode not in ("database", "analytics"):
            raise ValueError("`mode` must be either 'database' or 'analytics'")
```